ruby/lib/transformer/syskit/extensions/system_network_generator.rb

```diff
@@ -59,39 +70,77 @@ def validate_abstract_network
-                    SystemNetworkGeneratorExtension.validate_frame_selection_consistency_through_inputs(task)
+                    SystemNetworkGeneratorExtension
+                        .validate_frame_selection_consistency_through_inputs(
+                            task, error_handler
+                        )
                 end
             end
         end
 
-        def self.validate_frame_selection_consistency_through_inputs(task)
+        def self.validate_mismatched_annotated_input(task, error_handler)
             task.each_annotated_port do |task_port, task_frame|
                 next if !task_port.input? || !task_frame
-                task_port.each_frame_of_connected_ports do |other_port, other_frame|
-                    if other_frame != task_frame
-                        raise FrameSelectionConflict.new(
-                            task,
-                            task.model.find_frame_of_port(task_port),
-                            task_frame,
-                            other_frame)
-                    end
-                end
+
+                task_port.each_frame_of_connected_ports
+                         .find_all { |other_port, other_frame| other_frame != task_frame }
+                         .each do |port, frame|
+                             conflict = FrameSelectionConflict.new(
+                                task,
+                                task.model.find_frame_of_port(task_port),
+                                task_frame,
+                                frame
+                             )
+                             error_handler.register_resolution_failures_from_exception(
+                                [task], conflict
+                             )
+                         end
             end
+        end
+
+        def self.validate_mismatched_transform_input(task, error_handler)
             task.each_transform_port do |task_port, task_transform|
                 next if !task_port.input?
-                task_port.each_transform_of_connected_ports do |other_port, other_transform|
-                    if other_transform.from && task_transform.from && other_transform.from != task_transform.from
-                        task_local_name = task.model.find_transform_of_port(task_port).from
-                        raise FrameSelectionConflict.new(task, task_local_name,
-                                                         task_transform.from, other_transform.from)
-                    elsif other_transform.to && task_transform.to && other_transform.to != task_transform.to
-                        task_local_name = task.model.find_transform_of_port(task_port).to
-                        raise FrameSelectionConflict.new(task, task_local_name,
-                                                         task_transform.to, other_transform.to)
+
+                from_connections_mismatch = task_port
+                    .each_transform_of_connected_ports.to_a
```

Comment:
Why the `to_a` ? `find_all` is part of Enumerable

Reply:
I assume this is a leftover from before I added the #enum_for